intellij/test/junit/saros/intellij/filesystem/IntelliJReferencePointManagerTest.java

```diff
@@ -0,0 +1,186 @@
+    Assert.assertEquals(module, module2);
+  }
+
+  @Test
+  public void testPairPutIfAbsent() {
+    Module module = createModule("Module1");
+    intelliJReferencePointManager.putIfAbsent(referencePoint, module);
+    Module module2 = intelliJReferencePointManager.getModule(referencePoint);
+
+    Assert.assertNotNull(module2);
+    Assert.assertEquals(module, module2);
+  }
+
+  @Test
+  public void testGetResource() {
+    VirtualFile resource = EasyMock.createMock(VirtualFile.class);
+    EasyMock.replay(resource);
+
+    IPath relativePath = createReferencePointPath("./path/to/file");
+
+    Module module = createModule("Module1", resource, relativePath.toString());
+
+    VirtualFile moduleRoot = createModuleRoot(relativePath.toString(), resource);
+
+    // This is needed for mocking static stuff
+    createModuleRootManager(moduleRoot, module, resource);
+
+    IReferencePoint referencePoint = IntelliJReferencePointManager.create(module);
+
+    intelliJReferencePointManager.putIfAbsent(referencePoint, module);
+
+    VirtualFile file = intelliJReferencePointManager.getResource(referencePoint, relativePath);
+    Assert.assertNotNull(file);
+    Assert.assertEquals(resource, file);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testGetResourceFromEmptyIntelliJReferencePointManager() {
+    VirtualFile resource = EasyMock.createMock(VirtualFile.class);
+    EasyMock.replay(resource);
+
+    IPath relativePath = createReferencePointPath("./path/to/file");
+
+    intelliJReferencePointManager.getResource(referencePoint, relativePath);
+  }
+
+  @Test
+  public void testGetModuleRoot() {
+    VirtualFile moduleRootCopy = EasyMock.createMock(VirtualFile.class);
+    EasyMock.replay(moduleRootCopy);
+
+    VirtualFile moduleRoot =
+        createModuleRoot(createReferencePointPath("Module1").toString(), moduleRootCopy);
+
+    Module module =
+        createModule("Module1", moduleRoot, createReferencePointPath("Module1").toString());
+
+    // This is needed for mocking static stuff
+    createModuleRootManager(moduleRoot, module, moduleRootCopy);
+
+    IReferencePoint referencePoint = IntelliJReferencePointManager.create(module);
+
+    intelliJReferencePointManager.putIfAbsent(referencePoint, module);
+
+    VirtualFile file =
+        intelliJReferencePointManager.getResource(
+            referencePoint, createReferencePointPath("Module1"));
+    Assert.assertNotNull(file);
+    Assert.assertEquals(moduleRootCopy, file);
+  }
+
+  private IPath createReferencePointPath(String path) {
+    return IntelliJPathImpl.fromString(path);
+  }
+
+  private VirtualFile createModuleRoot(String pathToFile, VirtualFile resource) {
+    VirtualFile moduleRoot = EasyMock.createMock(VirtualFile.class);
+    EasyMock.expect(moduleRoot.findFileByRelativePath(pathToFile)).andStubReturn(resource);
+
+    EasyMock.replay(moduleRoot);
+
+    return moduleRoot;
+  }
+
+  @PrepareForTest(ModuleRootManager.class)
```

Comment:
Why do you annotate PrepareForTest (with ModuleRootManager) on this method and on class level?

Reply:
Because I need PowerMock for mocking static methods. I use the `ModuleRootManager` in `IntelliJReferencePointManager` and the static method `getInstance` is called their for getting the instance of `ModuleRootManager`. 
If I use PowerMock for mocking classes with static methods, I have to mark the test class with the Annotation `PrepareForTest `

Reply:
My questions was just: Why do you use the annotation also on this method not only on class level (you use the annotation also in line 22). Is this necessary?

Reply:
Ok, I was confused, because I thought you meant the Annotation (line 22). The annotation over the method `createModuleRootManager` is unnecessary and I removed this.